core/core_to_map_std/src/unstable/security.rs

```diff
@@ -107,13 +126,77 @@ impl MapInterpreterSecurityMisconfiguredError {
     provider_json: &ProviderJson,
-    security_values: &SecurityValuesMap,
+    map_security: &HostValue,
 ) -> Result<SecurityMap, PrepareSecurityMapError> {
     let security_schemes = match &provider_json.security_schemes {
         Some(security_schemes) => security_schemes,
         None => return Ok(SecurityMap::new()),
     };
 
+    let security_values = match &map_security {
+        HostValue::Object(obj) => {
+            let mut result = SecurityValuesMap::new();
+
+            for (id, config) in obj.iter() {
+                if let HostValue::Object(obj) = config {
+                    let security_value: SecurityValue;
+
+                    if obj.contains_key("apikey") {
+                        security_value = SecurityValue::ApiKey {
+                            apikey: match obj.get("apikey") {
+                                Some(HostValue::String(str)) => str.to_owned(),
+                                _ => {
+                                    unreachable!(
+                                        "Schema validation should ensure there is String value for apikey field."
+                                    );
+                                }
+                            },
+                        }
+                    } else if obj.contains_key("username") {
+                        security_value = SecurityValue::Basic {
+                            username: match obj.get("username") {
+                                Some(HostValue::String(str)) => str.to_owned(),
+                                _ => {
+                                    unreachable!(
+                                        "Schema validation should ensure there is String value for username field."
+                                    );
+                                }
+                            },
+                            password: match obj.get("password") {
+                                Some(HostValue::String(str)) => str.to_owned(),
+                                _ => {
+                                    unreachable!(
+                                        "Schema validation should ensure there is String value for password field."
+                                    );
+                                }
+                            },
+                        }
+                    } else if obj.contains_key("token") {
+                        security_value = SecurityValue::Bearer {
+                            token: match obj.get("token") {
+                                Some(HostValue::String(str)) => str.to_owned(),
+                                _ => {
+                                    unreachable!(
+                                        "Schema validation should ensure there is String value for token field."
+                                    );
+                                }
+                            },
+                        }
+                    } else {
+                        continue;
```

Comment:
question: shouldn't this also be `unreachable!`?

Reply:
I guess it should.